core/forage-core-common/pom.xml

```diff
@@ -27,6 +27,28 @@
             <artifactId>langchain4j-core</artifactId>
         </dependency>
         
+        <!-- Runtime dependencies -->
+        <dependency>
+            <groupId>org.springframework</groupId>
+            <artifactId>spring-core</artifactId>
+            <version>${spring-framework.version}</version>
+        </dependency>
+        <dependency>
+            <groupId>org.springframework</groupId>
+            <artifactId>spring-context</artifactId>
+            <version>${spring-framework.version}</version>
+        </dependency>
+        <dependency>
+            <groupId>org.eclipse.microprofile.config</groupId>
+            <artifactId>microprofile-config-api</artifactId>
+            <version>${microprofile.version}</version>
+        </dependency>
+        <dependency>
+            <groupId>io.smallrye.config</groupId>
+            <artifactId>smallrye-config</artifactId>
+            <version>${smallrye.version}</version>
+        </dependency>
+
```

Comment:
Keep in mind that this is bringing spring deps also when we export to Quarkus. I am a bit concerned w/ this ...

Reply:
I've thought through this a bit and if we're going to use the Spring / Quarkus programattic environments (vs. loading application.properties files), I don't think there's a way of removing the maven dependencies.      Maven's profiles would allow us to not load a set of dependencies at build time, but we would need to determine the environment at forage runtime.